src/libp2p.js

```diff
@@ -0,0 +1,330 @@
+'use strict'
+
+const Libp2p = require('libp2p')
+const TCP = require('libp2p-tcp')
+const Bootstrap = require('libp2p-bootstrap')
+const MPLEX = require('libp2p-mplex')
+const SECIO = require('libp2p-secio')
+const KadDHT = require('libp2p-kad-dht')
+const pullToStream = require('pull-stream-to-stream')
+const PeerBook = require('peer-book')
+const PeerInfo = require('peer-info')
+const PeerID = require('peer-id')
+const path = require('path')
+const multiaddr = require('multiaddr')
+
+/**
+ * Creates a PeerInfo from scratch, or via the supplied private key
+ * @param {string} privateKey Path to private key
+ * @returns {Promise} Resolves the created PeerInfo
+ */
+const getPeerInfo = (privateKey) => {
+  return new Promise((resolve, reject) => {
+    if (!privateKey) {
+      PeerInfo.create((err, peerInfo) => {
+        if (err) {
+          return reject(err)
+        }
+        resolve(peerInfo)
+      })
+    } else {
+      PeerID.createFromPrivKey(path.resolve(privateKey), (err, peerId) => {
+        if (err) {
+          return reject(err)
+        }
+        resolve(new PeerInfo(peerId))
+      })
+    }
+  })
+}
+
+class PeerRouting {
+  constructor (routing) {
+    this._routing = routing
+  }
+
+  /**
+   *
+   * @param {PeerID} id The peer to find
+   * @param {object} options
+   * @param {number} options.maxTimeout How long the query should run
+   * @returns {Promise}
```

Comment:
document what promise resolves to? There are some function here that document that, while others do not

(same below)